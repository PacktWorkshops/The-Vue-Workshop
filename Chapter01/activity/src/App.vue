<template>
  <div class="container">
    <h2>Shopping list</h2>
    <div class="user-input">
      <input
        placeholder="Press enter to add new item"
        v-model="input"
        @keyup.enter="addItem"
        ref="input"
      /><button @click="addItem">Add item</button>
    </div>

    <ul v-if="shoppingList">
      <li v-for="(item, i) in shoppingList" :key="i" class="item"
        ><span>{{ item }}</span>
        <button class="button--remove" @click="deleteItem(i)">Remove</button>
      </li>
    </ul>
    <br />
    <button class="button--delete" @click="deleteItem()">Delete all</button>
  </div>
</template>

<script>
export default {
  data() {
    return {
      input: '',
      shoppingList: [],
    }
  },
  methods: {
    addItem() {
      // Don't allow adding to the list if empty
      if (!this.input) return
      this.shoppingList.push(this.input)
      // Clear the input after adding
      this.input = ''
      // Focus the input element again for quick typing!
      this.$refs.input.focus()
    },
    deleteItem(i) {
      this.shoppingList = i
        ? this.shoppingList.filter((item, x) => x !== i)
        : []
    },
  },
}
</script>

<style lang="scss">
@import 'styles/global';

$color-green: #4fc08d;
$color-grey: #2c3e50;

.container {
  max-width: 600px;
  margin: 80px auto;
}

.item {
  display: flex;
  justify-content: space-between;
}

// Type
.h2 {
  font-size: 21px;
}

.user-input {
  display: flex;
  align-items: center;
  padding-bottom: 20px;
  input {
    width: 100%;
    padding: 10px 6px;
    margin-right: 10px;
  }
}

.item {
  display: flex;
  align-items: center;
}

// Buttons
button {
  appearance: none;
  padding: 10px;

  font-weight: bold;
  border-radius: 10px;
  border: none;
  background: $color-grey;
  color: white;
  white-space: nowrap;

  + button {
    margin-left: 10px;
  }
}

.button--delete {
  display: block;
  margin: 0 auto;
  background: red;
}

.button--remove {
  background: none;
  color: red;
  text-transform: uppercase;
  font-size: 11px;
  align-self: flex-end;
}

ul {
  display: block;
  margin: 0 auto;
  padding: 30px;
  border: 1px solid rgba(0, 0, 0, 0.25);

  > li {
    color: $color-grey;
    margin-bottom: 4px;
  }
}
</style>
