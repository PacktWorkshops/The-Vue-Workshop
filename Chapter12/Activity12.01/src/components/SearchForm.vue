<template>
  <form
    @submit="onSubmit()"
    class="flex flex-row m-auto mb-10"
  >
    <input
      v-model="searchTerm"
      placeholder="Search"
      class="bg-white focus:outline-none focus:shadow-outline border
      border-gray-300 rounded py-2 px-4 flex
      appearance-none leading-normal"
      type="text"
    />
    <button
      type="submit"
      class="flex bg-blue-500 hover:bg-blue-700
      text-white font-semibold font-sm hover:text-white py-2 px-4 border
      border-blue-500 hover:border-transparent rounded"
    >
      Search
    </button>
  </form>
</template>
<script>
export default {
  data() {
    return {
      searchTerm: this.$route.query.q || ''
    }
  },
  methods: {
    onSubmit() {
      this.$router.push({
        path: '/',
        query: {
          q: this.searchTerm
        }
      })
    }
  }
}
</script>
