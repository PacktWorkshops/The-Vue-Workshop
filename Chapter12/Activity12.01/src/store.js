import Vue from 'vue'
import Vuex from 'vuex'

Vue.use(Vuex)

export default new Vuex.Store({
  state: {
    dismissedSubscribeBanner: false,
    acceptedCookie: false
  },
  mutations: {
    dismissSubscribeBanner(state) {
      state.dismissedSubscribeBanner = true
    },
    acceptCookie(state) {
      state.acceptedCookie = true
    }
  }
})
