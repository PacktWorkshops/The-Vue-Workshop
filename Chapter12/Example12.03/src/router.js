import Vue from 'vue'
import VueRouter from 'vue-router'

import PostList from './components/PostList.vue'
import SinglePost from './components/SinglePost.vue'

Vue.use(VueRouter)

const routes = [
  {
    path: '/',
    component: PostList
  },
  {
    path: '/:postId',
    component: SinglePost
  }
]

export default new VueRouter({
  routes
})
