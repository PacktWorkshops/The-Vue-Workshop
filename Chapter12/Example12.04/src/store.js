import Vue from 'vue'
import Vuex from 'vuex'

Vue.use(Vuex)

export default new Vuex.Store({
  state: {
    dismissedSubscribeBanner: false
  },
  mutations: {
    dismissSubscribeBanner(state) {
      state.dismissedSubscribeBanner = true
    }
  }
})
