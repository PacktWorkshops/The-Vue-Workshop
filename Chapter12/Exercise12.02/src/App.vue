<template>
  <div id="app" class="p-10">
    <div class="flex flex-col">
      <h2
        class="leading-loose pb-4 flex justify-center m-auto md:w-1/3 text-xl mb-8 font-bold text-gray-800 border-b"
      >
      The Vue.js Workshop Blog
      </h2>
      <PostList :posts="posts" />
    </div>
  </div>
</template>

<script>
import PostList from './components/PostList.vue'

export default {
  components: {
    PostList
  },
  data() {
    return {
      posts: [
        {
          title: 'Vue.js for React developers',
          description: 'React has massive popularity here are the key benefits of Vue.js over it.',
          tags: ['vue', 'react'],
        },
        {
          title: 'Migrating an AngularJS app to Vue.js',
          description: 'With many breaking changes, AngularJS developers have found it easier to retrain to Vue.js than Angular 2',
          tags: ['vue', 'angularjs']
        }
      ]
    }
  }
}
</script>
