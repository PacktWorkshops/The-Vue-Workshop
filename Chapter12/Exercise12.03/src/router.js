import Vue from 'vue'
import VueRouter from 'vue-router'

import PostList from './components/PostList.vue'
import SinglePost from './components/SinglePost.vue'
import TagPage from './components/TagPage.vue'

Vue.use(VueRouter)

const routes = [
  {
    path: '/',
    component: PostList
  },
  {
    path: '/:postId',
    component: SinglePost
  },
  {
    path: '/tags/:tagName',
    component: TagPage
  }
]

export default new VueRouter({
  routes
})
