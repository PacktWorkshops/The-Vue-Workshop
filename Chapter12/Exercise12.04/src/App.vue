<template>
  <div id="app" class="p-10">
    <NewsletterBanner />
    <div class="flex flex-col">
      <h2
        class="leading-loose pb-4 flex justify-center m-auto md:w-1/3 text-xl mb-8 font-bold text-gray-800 border-b"
      >
      The Vue.js Workshop Blog
      </h2>
      <router-view
        :posts="posts"
      />
      <CookieBanner />
    </div>
  </div>
</template>

<script>
import NewsletterBanner from './components/NewsletterBanner.vue'
import CookieBanner from './components/CookieBanner.vue'

export default {
  components: {
    NewsletterBanner,
    CookieBanner
  },
  data() {
    return {
      posts: [
        {
          slug: 'vue-react',
          title: 'Vue.js for React developers',
          description: 'React has massive popularity here are the key benefits of Vue.js over it.',
          content:
            `React has massive popularity here are the key benefits of Vue.js over it.
            See the following table, we'll also look at how the is the content of the post.

            There's more, we can map React concepts to Vue and vice-versa.`,
          tags: ['vue', 'react'],
        },
        {
          slug: 'vue-angularjs',
          title: 'Migrating an AngularJS app to Vue.js',
          description: 'With many breaking changes, AngularJS developers have found it easier to retrain to Vue.js than Angular 2',
          content:
            `With many breaking changes, AngularJS developers have found it easier to retrain to Vue.js than Angular 2

            Vue.js keeps the directive-driven templating style while adding a component model.

            It's performant thanks to a great reactivity engine.`,
          tags: ['vue', 'angularjs']
        }
      ]
    }
  }
}
</script>
