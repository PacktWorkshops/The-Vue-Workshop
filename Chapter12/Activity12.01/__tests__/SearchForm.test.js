import {render, fireEvent} from '@testing-library/vue'
import SearchForm from '../src/components/SearchForm.vue'
import App from '../src/App.vue'
import router from '../src/router'
import store from '../src/store'

test('SearchForm should match expected HTML', () => {
  const {html} = render(SearchForm, {
    router,
    store
  })
  expect(html()).toMatchSnapshot()
})

test('SearchForm filter by keyword on submission', async () => {
  const {getByPlaceholderText, getByText, queryByText} = render(App, {
    router,
    store
  })
  expect(queryByText('Migrating an AngularJS app to Vue.js')).toBeTruthy()
  expect(queryByText('Vue.js for React developers')).toBeTruthy()
  await fireEvent.update(getByPlaceholderText('Search'), 'react')
  await fireEvent.click(getByText('Search'))
  expect(queryByText('Vue.js for React developers')).toBeTruthy()
  expect(queryByText('Migrating an AngularJS app to Vue.js')).toBeFalsy()
})
