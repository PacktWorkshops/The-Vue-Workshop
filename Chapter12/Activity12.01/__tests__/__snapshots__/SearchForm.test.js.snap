// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm should match expected HTML 1`] = `
<form class="flex flex-row m-auto mb-10"><input placeholder="Search" type="text" class="bg-white focus:outline-none focus:shadow-outline border
    border-gray-300 rounded py-2 px-4 flex
    appearance-none leading-normal"> <button type="submit" class="flex bg-blue-500 hover:bg-blue-700
    text-white font-semibold font-sm hover:text-white py-2 px-4 border
    border-blue-500 hover:border-transparent rounded">
    Search
  </button></form>
`;
